x-pack/plugins/code/public/reducers/symbol.ts

```diff
@@ -66,7 +66,7 @@ const generateStructureTree: (symbols: SymbolInformation[]) => any = symbols =>
     if (containerName === undefined) {
       return undefined;
     }
-    const regex = new RegExp(`^${containerName}[<(]?.*[>)]?$`);
+    const regex = new RegExp(`^${containerName}[[<(].*[>)]]?$`);
```

Comment:
I would suggest moving this function out of this file into a separate file. Then add test cases to protect it.

The input test cases should come from all these issues this PR tries to fix.

Reply:
Otherwise without the unit test, for regex, it's easy to fix one thing but break another.